### PR TITLE
DI-139 Get matching ods services

### PR DESCRIPTION
**Pull request by mabe13**
New PR and Branch of @SLClayton ticket to conform with PR and Branch naming conventions 

The codes and tests in this ticket still need a bit of refactoring, the lambda handler function also needs a bit of tidy up. Would sort this out but work has already start in a subquent Ticket (DI-168), which is being to address some of these issues. The remaining issue will be sort by other subquent tickets that effect the event proccess and a possible refactoring ticket.

## Link to JIRA Ticket

- https://nhsd-jira.digital.nhs.uk/browse/DI-139
- https://nhsd-jira.digital.nhs.uk/browse/DI-150
- https://nhsd-jira.digital.nhs.uk/browse/DI-129

## Description

Created the event_process lambda. And covers the functionality within tickets DI-139/150/129
 - Inputs Pharmacy data (NHS_entity)
 - Finds matching DoS services from DB
 - Creates change requests based on difference in website field
 - Sends requests to next lambda if optionally chosen to

This pull request is for a review as I am away next week, code has been tested locally but not built before.

## Type of change

- New feature (non-breaking change which adds functionality)

## Testing

Please tick the testing that has been completed

- [x] Unit tests
- [ ] Component tests

## Developer Checklist

- [x] I have performed a self-review of my own code
- [x] I have run the [code formatting checks](../README.md#code-quality)
- [x] I have run the [code quality checks](../README.md#code-quality)
- [x] New code meets [standards](https://nhsd-confluence.digital.nhs.uk/display/DI/DI+Ways+of+Working) agreed by the team
- [x] Unit test code coverage is at or above 80%
- [ ] New and existing unit tests pass locally with my changes
- [ ] Tests have added that prove my fix is effective or that my feature works (Component tests)
- [ ] I have made corresponding changes to the documentation
- [x] I have cleaned down my environment (if created)

## Code Reviewer Checklist

- [ ] I have run the unit tests and they run correctly
- [ ] I can confirm the changes have been tested or approved by a tester
- [x] I can confirm no remaining infrastructure is left over from this branch

### CODE REVIEW

**Review by JackPlowman — APPROVED** [review]
LGTM 👍, Happy enough with the PR for now, will refactor in DI-168 branch/PR.

File: deployment/serverless.yml [hunk 1]
```diff
@@ -113,6 +106,31 @@ functions:
...
+  event-processor:
+    image: ${env:AWS_SAME_ACCOUNT_DOCKER_REGISTRY}/event-processor:${env:VERSION}
+    architecture: x86_64
+    name: ${env:PROJECT_ID}-${sls:stage}-event-processor
+    description: ${sls:stage} Event Processor lambda
+    memorySize: 256
+    timeout: 120
+    role: arn:aws:iam::${env:AWS_ACCOUNT_ID}:role/${env:TF_VAR_event_processor_role_name}
+    tags:
+      PYTHON_VERSION: ${env:PYTHON_VERSION}
```
Comment (JackPlowman) [inline deployment/serverless.yml:118]: Needs to be changed to 
``` suggestion 
      PYTHON_VERSION: ${env:SERVERLESS_PYTHON_VERSION_TAG}
```

**Review by JackPlowman — CHANGES_REQUESTED** [review]
This small change needs to be made as it breaks the deployment make targets

**Review by JackPlowman — APPROVED** [review]
LGTM 👍, Happy enough with the PR for now, will refactor in DI-168 branch/PR.

**Review by mik3ola — APPROVED** [review]

**Review by JackPlowman — APPROVED** [review]
LGTM 👍, Happy enough with the PR for now, will refactor in DI-168 branch/PR.

**Pull request merged**